miden-crypto/src/hash/algebraic_sponge/rescue/arch/x86_64_avx2.rs

```diff
@@ -180,7 +180,7 @@ unsafe fn add_small(
 ) -> (__m256i, __m256i, __m256i) {
     unsafe {
         let res_wrapped_s = map3!(_mm256_add_epi64, x_s, y);
-        let mask = map3!(_mm256_cmpgt_epi32, x_s, res_wrapped_s);
+        let mask = map3!(_mm256_cmpgt_epi64, x_s, res_wrapped_s);
```

Comment:
The fix is correct, but we might want to add a targeted test for overflow at 32-bit boundaries to prevent regression. The current 5M iteration test should catch most issues, but a specific test with values that trigger overflow in the upper 32 bits could provide extra confidence.